### PR TITLE
Fix bug in sleep & al. passing event loop.

### PR DESCRIPTION
**Pull request by jonathanverner**
Brython's `asyncio` implementation only has a single loop, so none of its methods accept a loop argument. However, some methods were copied from the CPython implementation and were not adapted to this change. This commit fixes that by removing the loop argument from all methods in `asyncio/__init__.py`.

### CODE REVIEW

File: www/src/Lib/asyncio/__init__.py [hunk 1]
```diff
@@ -50,31 +50,31 @@ def process_query():
...
     if isinstance(coro_or_future, Future):
         return coro_or_future
     elif iscoroutine(coro_or_future):
         return coro_or_future(*args)
     else:
         raise Exception('Expecting coroutine or Future got '+str(coro_or_future)+' instead.')
 
-def gather(*coros_or_futures, loop=None, return_exceptions=False):
+def gather(*coros_or_futures, return_exceptions=False):
     fut_list = [ensure_future(c, loop=loop) for c in coros_or_futures]
```
Comment (mementum) [inline www/src/Lib/asyncio/__init__.py]: `loop=loop`, but `loop` is no longer an argument.

Shouldn't this refer to the default event loop (defined above in the file) as in `loop=default_event_loop`?

**Comment by mementum** [comment]
Removing the `loop` argument would prevent others from, for example, subclassing `BrowserEventWindow`. Although I have no use case at hand right now, it may be that some scenarios require hooking into the methods of an event loop.

Just a thought.

File: www/src/Lib/asyncio/__init__.py (hunk 1 above)
Comment (jonathanverner) [inline www/src/Lib/asyncio/__init__.py]: Good catch !, will update the pull request tomorrow.

**Comment by jonathanverner** [comment]
@mementum re single event loop: I think that even in CPython it is expected that only a single event loop will be running at any given point in time --- as far as I understand, there is no reason to run two event loops in parallel. Now in the browser, the browser event loop is always running, so if the above reasoning is correct, then there is no need having different eventloops. I don't currently see a use case for modifying the implementation of the event loop (which is really simple, just a thin wrapper around setTimeout & al.) and removing the possibility simplifies the code --- no need to pass the loop variable around everywhere. But if you come up with a valid use case, I can go back and add the support for different event-loop implementations back.

**Comment by PierreQuentel** [comment]
Thanks again to both of you for your intense work on Brython !

I will be on holidays next week, back on August 28.

**Comment by mementum** [comment]
@PierreQuentel Enjoy

The idea was not to have 2 event loops, but that is probably the end result, because there is a `default_event_loop`. One would need to first override this, for it to make sense.

At the end of the day and given the execution environment is much better for sure to stick to reasonable defaults and default behavior

**Pull request merged**